packages/nimble-components/src/table-column/enum-base/models/table-column-enum-base-validator.ts

```diff
@@ -0,0 +1,76 @@
+import { ColumnValidator } from '../../base/models/column-validator';
+import type { Mapping } from '../../../mapping/base';
+import type { MappingKey } from '../../../mapping/base/types';
+import type { MappingKeyType } from '../types';
+import { resolveKeyWithType } from './mapping-key-resolver';
+
+export const enumBaseValidityFlagNames = [
+    'invalidMappingKeyValueForType',
+    'unsupportedMappingType',
+    'duplicateMappingKey',
+    'missingKeyValue',
+    'missingLabelValue'
+] as const;
+
+/**
+ * Validator for TableColumnEnumText. Implementations MUST include enumBaseValidityFlagNames in validity flag names set.
+ */
+export abstract class TableColumnEnumBaseValidator<
+    ValidityFlagNames extends readonly string[]
+> extends ColumnValidator<
+    typeof enumBaseValidityFlagNames | ValidityFlagNames
+    > {
+    public validate(
+        supportedMappingElements: readonly (typeof Mapping)[],
```

Comment:
Do we even expect this set to be dynamic for a particular concrete column type? If not, this seems like a strange API, where we could instead provide this as a constructor parameter, and just have this state available on the class instance, instead of having to always pass some const as a parameter to a method.